### ISSUE TITLE
[Windows] rotki still shows available update after auto-updating. Backend-frontend mismatch

### ISSUE THREAD

**Issue by Linus045** [rotki/rotki#3944]
## Problem Definition
TLDR at the end.

After starting Rotki it asked me to download an available Update from version 1.23.0 (current)-> 1.23.1 (new).
I let it download and waited a bit, nothing seemed to be happening afterwards.
The update button in the top left corner was still showing and after pressing it and clicking "download" again it showed this message:

![image](https://user-images.githubusercontent.com/13884784/149639047-cd3203f9-fc68-4b64-9ff5-c23eb5668eb0.png)

I restarted the process and opened it again, after logging in the problem was still there:
![image](https://user-images.githubusercontent.com/13884784/149639072-802045d1-81e5-435a-a2e0-cbe108b7fb52.png)

After logging out of windows completly and back in (to make sure no other process was running) and starting rotki again it still showed the same update button and version number under help (see img above).

To fix the issue i manually renamed the old version that was still inside the resources directory:
![image](https://user-images.githubusercontent.com/13884784/149639122-d4e5a32a-cec0-494d-b38a-5c299f193cb7.png)

After restarting rotki now it correctly started the new version 1.23.1:
![image](https://user-images.githubusercontent.com/13884784/149639146-0af5437b-9f72-4722-b4c7-32a04f9ecafd.png)

tldr:
The old .exe wasn't removed after downloading/installing the update.
This could be a mistake on my part for closing the program before it was fully done, but I didn't receive any message while closing and (I'm pretty sure) there was no task running when i decided to close the program after trying to update.

## Logs
rotki_electron.log is attached (taken after the problem was fixed by removing the old .exe)
rotkehlchen.log file was completly empty for me (probably because the debug setting was disabled).

[rotki_electron.log](https://github.com/rotki/rotki/files/7875987/rotki_electron.log)

### System Description
Windows
Using Electron App
No Docker 

Operating system:  Windows
Rotki version: 1.23.0 -> 1.23.1

**Comment by LefterisJP** [comment]
This was quite a curious case that @Linus045 came across. It seems as if the self-update did not finish and was stopped in the middle, somehow not deleting the old backend binary but updating the frontend.

So ended up with both backend binaries in the `c:\Users\Linus Willms\AppData\Local\Programs\rotki\resources\rotkehlchen_py_dist\` directory.

And the code of the frontend tries to find the binary by taking the first (and it assumes only) binary in that directory:

https://github.com/rotki/rotki/blob/b94836dee65fe2619c648717552525c141a71319/frontend/app/src/py-handler.ts#L379-L412

So in this edge case it gets the old, and wrong binary causing a mismatch in version between frontend and backend.

Solution for the user is easy (delete the old binary)

But @kelsos any idea how this can happen? Botched/not finished update? Perhaps we can improve the backend binary searching logic of the frontend to find the latest version one if there are multiple?

**Comment by kelsos** [comment]
The part about pressing the download button for a second time and seeing an error should be fixed by #3954 but it will only work for users upgrading the next patch release.

As for the second part with the old binary remaining, the logs seem to confirm that what happened is that the old binary was not properly terminated by taskkill but continued running during the update.

The most logical explanation would be that since the file was still in use could not be deleted by the updater and thus remained in the directory. 

With #3959 at least there will be a helpful message when this happens.

As for the root cause, I have no idea how to replicate this, it could be a timing issue, or it could be anything.

From the logs, I can see that 3 processes were running during the update. In windows, the backend starts 2 processes, which means that for some reason one of the two processes from the previous session was terminated while the other kept running.

**Comment by LefterisJP** [comment]
I think the work you did at #3959 is enough to close the issue. If another issue with more info on something pops up we will just open another one.
